daemon/daemon_unix.go

```diff
@@ -903,6 +905,14 @@ func (daemon *Daemon) initNetworkController(config *config.Config, activeSandbox
 	}
 
 	// Set HostGatewayIP to the default bridge's IP  if it is empty
+	setHostGatewayIP(daemon, controller)
+
+	return controller, nil
+}
+
+/* This is a function that runs a block of code initially inside of initNetworkController
+This is a potential fix to an issue #42753 relating to containers not receiving host bridge addresses */
+func setHostGatewayIP(daemon *Daemon, controller libnetwork.NetworkController) {
 	if daemon.configStore.HostGatewayIP == nil && controller != nil {
```

Comment:
Perhaps instead of passing the whole daemon, we could make it get just the config-store. Looks like the nil-check for `controller` is also redundant (because at this point, we should always have a controller?)

Perhaps also change the `if` to use an early return, so that we don't need the nested ifs;

```go
// setHostGatewayIP sets cfg.HostGatewayIP to the default bridge's IP  if it is empty
func setHostGatewayIP(config *config.Config, controller libnetwork.NetworkController) {
	if config.HostGatewayIP != nil {
		return
	}
	if n, err := controller.NetworkByName("bridge"); err == nil {
	...
```

Reply:
Not related to your change, but It's actually odd that we're passing `config` as argument to `initNetworkController()`, because it's already available through `daemon`. We should probably change that (in a follow-up PR) to _either_ remove `daemon` _or_ the argument;

https://github.com/moby/moby/blob/9e7bbdb9ba53440cfc67e2e1de89e35e44c43ddf/daemon/daemon.go#L475